src/App.js

```diff
@@ -1,10 +1,78 @@
-
+import "./App.css";
+import { Routes, Route } from "react-router-dom";
+import { useEffect } from "react";
+import { ToastContainer, toast } from "react-toastify";
+import "react-toastify/dist/ReactToastify.css";
+import {
+  SideBar,
+  History,
+  Home,
+  useData,
+  useAuth,
+  Login,
+  SignUp,
+  Profile,
+  VideoPage,
+  LikedVideosPage,
+  Playlist,
+} from "./components/export";
+import {
+  getVideoList,
+  getLikedVideos,
+  getHistoryVideos,
+  getUserPlaylist,
+} from "./Utils/apiRequest";
+import { PrivateRoute } from "./PrivateRoute";
 function App() {
+  const { userDetails, isUserloggedIn, userId } = useAuth();
+
+  const { dispatch } = useData();
+
+  useEffect(() => {
+    getVideoList(dispatch);
+  }, []);
+
+  useEffect(() => {
+    if (isUserloggedIn && userId) {
+      getLikedVideos(userId, dispatch);
+    }
+  }, [isUserloggedIn]);
+
+  useEffect(() => {
+    if (isUserloggedIn && userId) {
+      getHistoryVideos(userId, dispatch);
+    }
+  }, [isUserloggedIn]);
+
+  useEffect(() => {
+    if (userId) {
+      getUserPlaylist(userId, dispatch);
+    }
+  }, [isUserloggedIn]);
+
```

Comment:
converge these calls into a single call=> put all the functional calls into a single useEffect